src/modules/sensors/sensor_params_mag0.c

```diff
@@ -102,9 +102,11 @@ PARAM_DEFINE_INT32(CAL_MAG0_EN, 1);
  * @value 43 Pitch 90°, Yaw 180°
  * @value 44 Pitch 9°, Yaw 180°
  * @value 45 Pitch 45°
+ * @value 46 Roll 270°, Yaw 225°
```

Comment:
Add a value of 1000 here that means "custom". Then add three parameters for the angles (in degrees) for all three. Then wrap the rotate function (please don't add parameters into it, because those are runtime heavy) in the mag driver where it will either take one of the standard ones, or if you have set it to custom, where it will read these parameters and apply a full rotation.